presto-spi/src/main/java/com/facebook/presto/spi/predicate/DomainExpression.java

```diff
@@ -1,8 +1,112 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 package com.facebook.presto.spi.predicate;
 
-/**
- * Created by praveen-3307 on 13/4/17.
- */
-public class DomainExpression
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+
+public class DomainExpression<C>
+        extends TupleExpression<C>
 {
+    private final C column;
+    private final Domain domain;
+
+    @JsonCreator
+    public DomainExpression(@JsonProperty("column") C column, @JsonProperty("domain") Domain domain)
+    {
+        this.column = column;
+        this.domain = domain;
+    }
+
+    @JsonProperty
+    public C getColumn()
+    {
+        return column;
+    }
+
+    @JsonProperty
+    public Domain getDomain()
+    {
+        return domain;
+    }
+
+    @Override
+    public Map<C, NullableValue> extractFixedValues()
+    {
+        Map<C, NullableValue> map = new HashMap();
```

Comment:
Well for Presto-spi module we should have minimal dependencies...so used it....If we need to use `ImmutableMap.Builder `then I think we should add guava as dependency. As of now guava library is used only for testing spi.